pkg/providers/vsphere/virtualmachine/configspec.go

```diff
@@ -115,6 +115,12 @@ func CreateConfigSpec(
 		}
 	}
 
+	// Initially set the guest ID in ConfigSpec to ensure VM is created with the expected guest ID.
+	// Afterwards, only update it if the VM spec guest ID differs from the existing ConfigInfo.
+	if guestID := string(vmCtx.VM.Spec.GuestID); guestID != "" {
```

Comment:
Do we ever want to honor an existing GuestID in the class ConfigSpec? If not, it should be cleared in `SanitizeVMClassConfigSpec()`